write-fonts/src/tables/cmap.rs

```diff
@@ -65,46 +107,84 @@ impl Cmap {
         let mut end_code = Vec::new();
         let mut start_code = Vec::new();
         let mut id_deltas = Vec::new();
+        let mut id_range_offsets = Vec::new();
+        let mut glyph_id_array = Vec::new();
+        let mut pending_range_offset = Vec::new();
 
         let mut mappings: Vec<_> = mappings.into_iter().collect();
         mappings.sort();
 
-        let mut prev = (u16::MAX - 1, u16::MAX - 1);
+        let mut run = CmapRun::new();
         for (cp, gid) in mappings.into_iter() {
-            let cp = (cp as u32).try_into().unwrap();
-            let next_in_run = (
-                prev.0.checked_add(1).unwrap(),
-                prev.1.checked_add(1).unwrap(),
-            );
-            let current = (cp, gid.to_u16());
-            // Codepoint and gid need to be continuous
-            if current != next_in_run {
+            let cp = cp as u32;
+            let delta = gid.to_u16() as i64 - cp as i64;
```

Comment:
> Oh if I just missed the mod 65536 line that's easier which would be nice.

yes, here you should do % 65536 just like fonttools does, then it fill fit:

```python
idDelta.append((indices[0] - startCode[i]) % 0x10000)
```

https://github.com/fonttools/fonttools/blob/3b9a73ff8379ab49d3ce35aaaaf04b3a7d9d1655/Lib/fontTools/ttLib/tables/_c_m_a_p.py#L1003

Reply:
Done

Reply:
Hm. I still get overflows.

Reply:
(153 - 0xa78b) % 0x10000) gives me 22798 in python. no idea

Reply:
TIL Rust % means remainder and Python % means modulus.

Reply:
OK, so the simple solution here is to do what Python means by %. Good catch.